mantle/cosa/schema.go

```diff
@@ -0,0 +1,78 @@
+package cosa
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/pkg/errors"
+
+	schema "github.com/xeipuuv/gojsonschema"
+)
+
+var (
+	// JSON Schema document. Default the generated Schema.
+	SchemaJSON = generatedSchemaJSON
+)
+
+func init() {
+	runtimeSchemaPath := os.Getenv("COSA_META_SCHEMA")
+	if runtimeSchemaPath != "" {
+		f, err := os.Open(runtimeSchemaPath)
+		if err != nil {
+			panic(errors.Wrapf(err, "failed to open schema file %s", runtimeSchemaPath))
+		}
+		defer f.Close()
+
+		if err := SetSchemaFromFile(f); err != nil {
+			panic(errors.Wrapf(err, "failed to read in schema file %s", runtimeSchemaPath))
+		}
+	}
+}
+
+// SetSchemaFromFile sets the validation JSON Schema
+func SetSchemaFromFile(r io.Reader) error {
+	if r == nil {
+		return errors.New("schema input is invalid")
+	}
+	in, err := ioutil.ReadAll(r)
+	if err != nil {
+		SchemaJSON = string(in)
+	}
+	return nil
+}
+
+// Validator checks the build against the schema.
+func (build *Build) Validator() (bool, []error) {
+	var e []error
+	data, err := json.Marshal(build)
+	if err != nil {
+		return false, append(e, err)
+	}
+	if len(data) == 0 {
+		return false, append(e,
+			errors.New("build data is empty"),
+		)
+	}
+
+	result, err := schema.Validate(
+		schema.NewStringLoader(SchemaJSON),
+		schema.NewStringLoader(string(data)),
+	)
+
+	if err != nil {
+		return false, append(e, err)
+	}
+
+	if result.Valid() {
+		return true, e
```

Comment:
`e` is always `nil` in this case. Might as well explicitly return that for readability.

Reply:
Fixed.